diff -pruN old/src/fr-command-7z.c new/src/fr-command-7z.c
--- old/src/fr-command-7z.c	2011-04-26 17:29:35.000000000 +0300
+++ new/src/fr-command-7z.c	2024-03-17 15:47:55.707881197 +0200
@@ -105,7 +105,7 @@ list__process_line (char     *line,
 			strncpy (version, ver_start, ver_len);
 			version[ver_len] = 0;
 
-			if (strcmp (version, "4.55") < 0)
+			if ((strcmp (version, "4.55") < 0) && (ver_len > 1) && (version[1] == '.'))
 				p7z_comm->old_style = TRUE;
 			else
 				p7z_comm->old_style = FALSE;
