diff -pruN old/src/libgdl/gdl-dock-item-grip.c new/src/libgdl/gdl-dock-item-grip.c
--- old/src/libgdl/gdl-dock-item-grip.c	2020-04-10 00:37:16.000000000 +0300
+++ new/src/libgdl/gdl-dock-item-grip.c	2023-01-14 00:06:55.204159201 +0200
@@ -418,7 +418,10 @@ gdl_dock_item_grip_realize (GtkWidget *w
         GdkWindowAttr  attributes;
         GdkCursor     *cursor;
 
-        g_return_if_fail (grip->_priv->label != NULL);
+        //g_return_if_fail (grip->_priv->label != NULL);
+        if (grip->_priv->label == NULL) {
+           gdl_dock_item_grip_set_label (grip, gdl_dock_item_create_label_widget(grip) );
+        }
 
         gtk_widget_get_allocation (grip->_priv->label, &allocation);
 
