<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN"
    "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
 <head>
  <style type="text/css">
   body {
    font-family: Verdana, sans-serif;
    font-size: 85%;
    padding: 2em;
   }
   table {
    border: solid #CAD4E9 1px;
    font-size: 85%;
   }
   td { padding: 6px; }
   tr.header { background-color: #CAD4E9; }
   tr.odd { background-color: #ECF0F7; }
   tr.even { background-color: #F7F9FC; }
   a { color: black; }
  </style>
  <title>CRUX ports</title>
  <meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1" />
 </head>
<body>
 <h2>CRUX ports</h2>
  <table width="100%" cellspacing="0">
  <tr class="header"><td colspan="4">
  <strong>Signify public key:</strong> RWToVVsIkzFIZebnqOd9r7OwjuCFjOdTN2ZMnVLflPnOV4ZBRwA/a1I6
  </td></tr>
   <tr class="header"><td><strong>Port</strong></td><td><strong>Version</strong></td><td><strong>Description</strong></td><td><strong>Last modified</strong></td></tr>
   <tr class="odd"><td><a href="https://sourceforge.net/projects/acpid2/files/">acpid</a></td><td><a href="./acpid/">2.0.34-1</a></td><td>ACPI daemon</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://sourceforge.net/projects/aiksaurus/files/aiksaurus/">aiksaurus</a></td><td><a href="./aiksaurus/">1.2.1-1</a></td><td>CLI/GTK thesaurus</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://ormbunkar.se/aliview/downloads/linux/">aliview</a></td><td><a href="./aliview/">1.30-1</a></td><td>Alignment viewer and editor</td><td>2026-02-10</td></tr>
   <tr class="even"><td><a href="https://sourceforge.net/projects/argtable/files/argtable/">argtable</a></td><td><a href="./argtable/">2.13-1</a></td><td>ANSI C library for parsing command line arguments</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://github.com/neeeeow/Bluecurve">bluecurve</a></td><td><a href="./bluecurve/">3.0.0-5</a></td><td>Bluecurve GTK 2/3/4 theme</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://github.com/BioPP/bpp-core/">bpp-core</a></td><td><a href="./bpp-core/">3.0.0_91d7e9b-1</a></td><td>Bio++ Core Library.</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://github.com/BioPP/bpp-phyl-omics/">bpp-phyl-omics</a></td><td><a href="./bpp-phyl-omics/">3.0.0_c160a67-1</a></td><td>Bio++ Phyl Omics Library.</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://github.com/BioPP/bpp-phyl/">bpp-phyl</a></td><td><a href="./bpp-phyl/">3.0.0_af18d54-1</a></td><td>Bio++ Phyl Library.</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://github.com/BioPP/bpp-popgen/">bpp-popgen</a></td><td><a href="./bpp-popgen/">3.0.0_ee574f8-1</a></td><td>Bio++ PopGen Library.</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://github.com/BioPP/bpp-seq-omics/">bpp-seq-omics</a></td><td><a href="./bpp-seq-omics/">3.0.0_3e269fd-1</a></td><td>Bio++ Seq Omics Library.</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://github.com/BioPP/bpp-seq/">bpp-seq</a></td><td><a href="./bpp-seq/">3.0.0_ae516ec-1</a></td><td>Bio++ Seq Library.</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="http://biopp.univ-montp2.fr/">bpp1-numcalc</a></td><td><a href="./bpp1-numcalc/">1.8.0-2</a></td><td>Bio++ 1 NumCalc Library</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="http://biopp.univ-montp2.fr/">bpp1-phyl</a></td><td><a href="./bpp1-phyl/">1.9.0-2</a></td><td>Bio++ 1 Phyl Library</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="http://biopp.univ-montp2.fr/">bpp1-seq</a></td><td><a href="./bpp1-seq/">1.7.0-2</a></td><td>Bio++ 1 Seq Library</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="http://biopp.univ-montp2.fr/">bpp1-utils</a></td><td><a href="./bpp1-utils/">1.5.0-2</a></td><td>Bio++ 1 Utils Library</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://github.com/BioPP/bppsuite/">bppsuite</a></td><td><a href="./bppsuite/">3.0.0_e6c912a-1</a></td><td>Bio++ Suite</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="http://bioinf.gen.tcd.ie/~faresm/software/software.html">caps-coev</a></td><td><a href="./caps-coev/">2.0-3</a></td><td>Coevolution Analysis using Protein Sequences</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://bio.tools/clustalo">clustalo</a></td><td><a href="./clustalo/">1.2.4-1</a></td><td>Clustal Omega</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://ftp.gnu.org/gnu/datamash/">datamash</a></td><td><a href="./datamash/">1.9-1</a></td><td>Basic numeric, textual and statistical operations</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="http://ftp.ebi.ac.uk/pub/software/vertebrategenomics/exonerate/">exonerate</a></td><td><a href="./exonerate/">2.4.0-1</a></td><td>Exonerate: a generic tool for sequence alignment</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="http://gbgoffice.info/">gbgoffice</a></td><td><a href="./gbgoffice/">1.4-1</a></td><td>Bgoffice dictionary frontend</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://molevol-ibe.csic.es/">gblocks</a></td><td><a href="./gblocks/">0.91b-1</a></td><td>Select blocks of evolutionarily conserved sites</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://gcc.gnu.org/gcc-5/">gcc5</a></td><td><a href="./gcc5/">5.5.0-1</a></td><td>The GNU Compiler Collection version 5.5.0</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://www.math.wustl.edu/~sawyer/geneconv/">geneconv</a></td><td><a href="./geneconv/">1.81a-1</a></td><td>Statistical Tests for Detecting Gene Conversion</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="http://www.gnumeric.org/">gnumeric</a></td><td><a href="./gnumeric/">1.12.59-1</a></td><td>Gnumeric: an open-source spreadsheet program</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="http://ftp.gnome.org/pub/GNOME/sources/goffice/">goffice</a></td><td><a href="./goffice/">0.10.59-2</a></td><td>A glib/gtk set of document centric objects and utilities</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://ftp.gnome.org/pub/GNOME/sources/gtkmm/">gtkmm2</a></td><td><a href="./gtkmm2/">2.24.5-1</a></td><td>C++ interface to GTK2.</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://github.com/HDFGroup/hdf5">hdf5</a></td><td><a href="./hdf5/">1.14.6-1</a></td><td>Hierarchical Data Format 5</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://hexchat.github.io/">hexchat</a></td><td><a href="./hexchat/">2.16.2-1</a></td><td>GTK+ based IRC client</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://developers.hp.com/hp-linux-imaging-and-printing/gethplip">hplip</a></td><td><a href="./hplip/">3.25.8-1</a></td><td>HP print/scan/fax support</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://github.com/veg/hyphy/">hyphy</a></td><td><a href="./hyphy/">2.5.93-1</a></td><td>Hypothesis Testing using Phylogenies</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://github.com/KDAB/KDSingleApplication">kdsingleapplication</a></td><td><a href="./kdsingleapplication/">1.2.0-1</a></td><td>KDAB's helper class for single-instance policy applications</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://doua.prabi.fr/software/lalnview">lalnview</a></td><td><a href="./lalnview/">3.0-3</a></td><td>Visualize local alignments (includes SIM)</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://download.gnome.org/sources/libgsf/">libgsf</a></td><td><a href="./libgsf/">1.14.55-1</a></td><td>GNOME Structured File Library</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://llvm.org/">llvm14</a></td><td><a href="./llvm14/">14.0.6-1</a></td><td>LLVM14 compiler backend</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://gitlab.com/sysimm/mafft/-/tags">mafft</a></td><td><a href="./mafft/">7.526-1</a></td><td>A multiple sequence alignment program</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://github.com/wjaguar/mtPaint/">mtpaint</a></td><td><a href="./mtpaint/">a50460b-1</a></td><td>simple GTK+ painting program</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://msa.sbc.su.se/cgi-bin/msa.cgi">mumsa</a></td><td><a href="./mumsa/">1.0-1</a></td><td>Assessment of the quality of multiple sequence alignments</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://www.drive5.com/muscle/downloads_v3.htm">muscle</a></td><td><a href="./muscle/">3.8.1551-1</a></td><td>MUltiple Sequence Comparison by Log-Expectation</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://github.com/rcedgar/muscle">muscle5</a></td><td><a href="./muscle5/">5.3-2</a></td><td>MUSCLE 5: Next-generation MUSCLE</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://ftp.ncbi.nlm.nih.gov/blast/executables/blast+/">ncbi-blast-plus</a></td><td><a href="./ncbi-blast-plus/">2.17.0-1</a></td><td>BLAST+ Command Line Applications</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://github.com/ncbi/datasets">ncbi-datasets</a></td><td><a href="./ncbi-datasets/">18.16.0-1</a></td><td>NCBI Datasets command-line tools</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://www.onlyoffice.com/download-desktop">onlyoffice</a></td><td><a href="./onlyoffice/">9.2.1-1</a></td><td>OnlyOffice Desktop Editors</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://github.com/OpenVPN/openvpn">openvpn</a></td><td><a href="./openvpn/">2.6.17-1</a></td><td>A Secure tunneling daemon</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://developer.apple.com/fonts/">otf-apple-sf-mono</a></td><td><a href="./otf-apple-sf-mono/">6.0.1.1726709071-1</a></td><td>Apple San Francisco mono font</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="http://www.bork.embl.de/pal2nal/">pal2nal</a></td><td><a href="./pal2nal/">14.1-1</a></td><td>Create a codon-based DNA alignment</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://github.com/abacus-gene/paml">paml</a></td><td><a href="./paml/">4.10.10_4c7902f-1</a></td><td>PAML: Phylogenetic Analysis by Maximum Likelihood</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="http://www.pdflib.com/download/pdflib-family/pdflib-lite-7/">pdflib-lite</a></td><td><a href="./pdflib-lite/">7.0.5p3-1</a></td><td>A library for generating PDF files</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://github.com/stephaneguindon/phyml/">phyml</a></td><td><a href="./phyml/">3.3.20250515-1</a></td><td>Phylogenetic estimation using Maximum Likelihood</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="http://gnome-look.org/content/show.php/Polar+Cursor+Theme?content=27913">polar-cursor-themes</a></td><td><a href="./polar-cursor-themes/">1.4-2</a></td><td>Polar cursor themes</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://github.com/ariloytynoja/prank-msa/tree/master/binaries">prank</a></td><td><a href="./prank/">251117-1</a></td><td>A probabilistic multiple alignment program</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://github.com/serge-sans-paille/beniget">python3-beniget</a></td><td><a href="./python3-beniget/">0.4.2.post1-1</a></td><td>A collection of Compile-time analyse on Python Abstract Syntax Tree(AST)</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://github.com/serge-sans-paille/gast">python3-gast</a></td><td><a href="./python3-gast/">0.6.0-1</a></td><td>A generic AST to represent Python3's Abstract Syntax Tree(AST).</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://pypi.org/project/joblib/">python3-joblib</a></td><td><a href="./python3-joblib/">1.5.2-1</a></td><td>Python functions as pipeline jobs</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://github.com/numba/llvmlite">python3-llvmlite</a></td><td><a href="./python3-llvmlite/">0.43.0-1</a></td><td>A Lightweight LLVM Python Binding for Writing JIT Compilers</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://github.com/numba/numba/">python3-numba</a></td><td><a href="./python3-numba/">0.61.2-1</a></td><td>Translates Python functions to optimized machine code</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://github.com/fatiando/pooch">python3-pooch</a></td><td><a href="./python3-pooch/">1.8.2-1</a></td><td>A friend to fetch your data files</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://github.com/pybind/pybind11/">python3-pybind11</a></td><td><a href="./python3-pybind11/">2.13.6-1</a></td><td>Seamless operability between C++11 and Python</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://github.com/lmcinnes/pynndescent">python3-pynndescent</a></td><td><a href="./python3-pynndescent/">0.5.13-1</a></td><td>Nearest neighbor descent for approximate nearest neighbors</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://github.com/serge-sans-paille/pythran">python3-pythran</a></td><td><a href="./python3-pythran/">0.18.0-1</a></td><td>A compiler for a Python subset</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://github.com/scikit-learn/scikit-learn">python3-scikit-learn</a></td><td><a href="./python3-scikit-learn/">1.7.1-1</a></td><td>A Python module for machine learning</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="http://www.scipy.org/">python3-scipy</a></td><td><a href="./python3-scipy/">1.16.1-1</a></td><td>Open-source software for mathematics, science, and engineering.</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://github.com/joblib/threadpoolctl">python3-threadpoolctl</a></td><td><a href="./python3-threadpoolctl/">3.6.0-1</a></td><td>Limit the number of threads</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://pypi.python.org/pypi/tinycss2">python3-tinycss2</a></td><td><a href="./python3-tinycss2/">1.5.1-1</a></td><td>A low-level CSS parser and generator for Python</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://pypi.org/project/tqdm/">python3-tqdm</a></td><td><a href="./python3-tqdm/">4.67.1-1</a></td><td>Fast, Extensible Progress Meter</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://github.com/lmcinnes/umap">python3-umap</a></td><td><a href="./python3-umap/">0.5.9.post2-1</a></td><td>Uniform Manifold Approximation and Projection in Python</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://sourceforge.net/projects/qt5ct/files/">qt5ct</a></td><td><a href="./qt5ct/">1.9-4</a></td><td>Qt5 Configuration Utility</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://www.opencode.net/trialuser/qt6ct/-/releases">qt6ct</a></td><td><a href="./qt6ct/">0.11-1</a></td><td>Qt6 configuration utility</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://www.opencode.net/trialuser/qt6gtk2/-/releases">qt6gtk2</a></td><td><a href="./qt6gtk2/">0.7-1</a></td><td>GTK+2.0 integration plugins for Qt6</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://posit.co/download/rstudio-desktop/">rstudio-desktop</a></td><td><a href="./rstudio-desktop/">2026.01.0+392-1</a></td><td>RStudio is an integrated development environment (IDE) for R</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="http://informagen.com/SA/">sa</a></td><td><a href="./sa/">1.7.2-1</a></td><td>Biologic sequences analysis</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="http://doua.prabi.fr/software/seaview">seaview</a></td><td><a href="./seaview/">5.1-1</a></td><td>Multiple sequence alignment and phylogeny GUI</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://bioinf.shenwei.me/seqkit/">seqkit</a></td><td><a href="./seqkit/">2.12.0-1</a></td><td>Atoolkit for FASTA/Q file manipulation</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://www.smplayer.info/en/downloads">smplayer</a></td><td><a href="./smplayer/">25.6.0-3</a></td><td>MPlayer/MPV frontend</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="ftp://ftp.pasteur.fr/pub/gensoft/projects/squizz/">squizz</a></td><td><a href="./squizz/">0.99d-1</a></td><td>A sequence/alignment format checker and converter</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://sourceforge.net/projects/sshpass/files/sshpass/">sshpass</a></td><td><a href="./sshpass/">1.10-1</a></td><td>Non-interactive ssh password auth files</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://www.strawberrymusicplayer.org/">strawberry</a></td><td><a href="./strawberry/">1.2.18-1</a></td><td>Strawberry music player</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://www.tcl.tk/">tcl</a></td><td><a href="./tcl/">8.6.16-1</a></td><td>The Tcl scripting language.</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://www.tcl.tk/">tk</a></td><td><a href="./tk/">8.6.16-1</a></td><td>A GUI toolkit for TCL</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://github.com/Ensembl/treebest">treebest</a></td><td><a href="./treebest/">347fa82-1</a></td><td>Tree Building guidEd by Species Tree</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://www.cgl.ucsf.edu/chimerax/download.html">ucsf-chimerax</a></td><td><a href="./ucsf-chimerax/">1.11.1-1</a></td><td>The next-generation molecular visualization program </td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://viber.com">viber</a></td><td><a href="./viber/">27.0.0.1-1</a></td><td>Viber for PC</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://vscodium.com/">vscodium</a></td><td><a href="./vscodium/">1.109.01000-1</a></td><td>Free/Libre Open Source Software Binaries of VS Code</td><td>2026-02-10</td></tr>
   <tr class="even"><td><a href="https://www.wps.cn/product/wpslinux#">wps-office</a></td><td><a href="./wps-office/">12.1.2.24722-1</a></td><td>WPS Office for Linux</td><td>2026-02-10</td></tr>
   <tr class="odd"><td><a href="https://www.jwz.org/xscreensaver/download.html">xscreensaver</a></td><td><a href="./xscreensaver/">6.14-2</a></td><td>XScreenSaver patched to work with electricsheep and rss-glx</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://github.com/xtensor-stack/xsimd/tags">xsimd</a></td><td><a href="./xsimd/">14.0.0-1</a></td><td>X Single Instruction, Multiple Data</td><td>2026-02-09</td></tr>
   <tr class="odd"><td><a href="https://github.com/yt-dlp/yt-dlp">yt-dlp</a></td><td><a href="./yt-dlp/">2026.02.04-1</a></td><td>CLI program to access YouTub and other data content</td><td>2026-02-09</td></tr>
   <tr class="even"><td><a href="https://zoom.us/download">zoom-linux</a></td><td><a href="./zoom-linux/">6.7.5.6891-1</a></td><td>Online conference</td><td>2026-02-10</td></tr>
   <tr class="odd"><td><a href="https://www.zotero.org/download/">zotero</a></td><td><a href="./zotero/">8.0.3-1</a></td><td>Zotero is a free, easy-to-use tool to help you collect, organize, cite, and share research</td><td>2026-02-09</td></tr>
  </table>
  <p><strong>89 ports</strong></p>
  <p><em>Generated by portspage 1.1 on 2026-02-10 21:43:35 UTC.</em></p>
  </body>
</html>