<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.1//EN"
    "http://www.w3.org/TR/xhtml11/DTD/xhtml11.dtd">

<html xmlns="http://www.w3.org/1999/xhtml">
 <head>
  <title>CRUX ports</title>
  <style type="text/css">
   body
   {
    font-family: Verdana, sans-serif;
    font-size: 85%;
    padding: 2em;
   }
   a
   {
    color: black;
   }
   table
   {
    border: solid #CAD4E9 1px;
    font-size: 85%;
   }
   td
   {
    padding: 6px;
   }
   tr.header
   {
    background-color: #CAD4E9;
   }
   tr.odd
   {
    background-color: #ECF0F7;
   }
   tr.even
   {
    background-color: #F7F9FC;
   }
  </style>
  <meta http-equiv="Content-Type" content="text/html; charset=ISO-8859-1" />
 </head>
 <body>
  <h2>CRUX ports</h2>
  <table width="100%" cellspacing="0">
  <tr class="header"><td colspan="4">
  <strong>Signify public key:</strong> RWTXrKv/sDjJcIMYDB881qk1+LzOUNwFJLmf7qtAeJC5L+B3gtAFeSIr
  </td></tr>
   <tr class="header"><td><b>Port</b></td><td><b>Version</b></td><td><b>Description</b></td><td><b>Last modified</b></td></tr>
   <tr class="odd"><td><a href="http://crystal.scb.uwa.edu.au/charlie/software/aline">aline</a></td><td><a href="./aline/">011208-1</a></td><td>A Sequence Alignment Editor and Viewer</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="http://www.ormbunkar.se/aliview/">aliview</a></td><td><a href="./aliview/">1.28-1</a></td><td>Alignment viewer and editor</td><td>2022-07-06</td></tr>
   <tr class="odd"><td><a href="https://github.com/narunlifescience/AlphaPlot">alphaplot</a></td><td><a href="./alphaplot/">1.02-1</a></td><td>Scientific data analysis and visualization</td><td>2022-07-15</td></tr>
   <tr class="even"><td><a href="http://amplicon.sourceforge.net/">amplicon</a></td><td><a href="./amplicon/">b09.1-1</a></td><td>Design PCR primers on aligned sequences</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="https://sites.google.com/site/cmzmasek/home/software/archaeopteryx">aptx</a></td><td><a href="./aptx/">1050-1</a></td><td>Archaeopteryx - visualise, analyse and edit phylogenetic trees</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="http://www.bioinformatics.org/genpak/">arka</a></td><td><a href="./arka/">0.11-1</a></td><td>A GUI for genpak</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="https://bedtools.readthedocs.io/">bedtools</a></td><td><a href="./bedtools/">2.30.0-1</a></td><td>A powerful toolset for genome arithmetic</td><td>2022-07-22</td></tr>
   <tr class="even"><td><a href="https://www.encodeproject.org/software/bedgraphtobigwig/">bigwig</a></td><td><a href="./bigwig/">4-1</a></td><td>Convert bedGraph to bigWig file</td><td>2022-07-22</td></tr>
   <tr class="odd"><td><a href="https://biopython.org/">biopython</a></td><td><a href="./biopython/">1.79-1</a></td><td>A set of Python tools for biological computation</td><td>2022-07-21</td></tr>
   <tr class="even"><td><a href="https://github.com/BenLangmead/bowtie/">bowtie</a></td><td><a href="./bowtie/">1.3.1-1</a></td><td>A short read aligner of DNA sequences</td><td>2022-07-21</td></tr>
   <tr class="odd"><td><a href="https://github.com/BenLangmead/bowtie2/">bowtie2</a></td><td><a href="./bowtie2/">2.4.5-1</a></td><td>A tool for aligning sequencing reads</td><td>2022-07-21</td></tr>
   <tr class="even"><td><a href="https://github.com/BioPP/bpp-seq/">bpp-core</a></td><td><a href="./bpp-core/">2.4.1-1</a></td><td>Bio++ Core Library.</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="https://github.com/BioPP/bpp-seq/">bpp-phyl</a></td><td><a href="./bpp-phyl/">2.4.1-1</a></td><td>Bio++ Phyl Library.</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="https://github.com/BioPP/bpp-seq/">bpp-phyl-omics</a></td><td><a href="./bpp-phyl-omics/">2.4.1-1</a></td><td>Bio++ Phyl Omics Library.</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="https://github.com/BioPP/bpp-seq/">bpp-popgen</a></td><td><a href="./bpp-popgen/">2.4.1-1</a></td><td>Bio++ PopGen Library.</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="https://github.com/BioPP/bpp-seq/">bpp-seq</a></td><td><a href="./bpp-seq/">2.4.1-1</a></td><td>Bio++ Seq Library.</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="https://github.com/BioPP/bpp-seq/">bpp-seq-omics</a></td><td><a href="./bpp-seq-omics/">2.4.1-1</a></td><td>Bio++ Seq Omics Library.</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="http://biopp.univ-montp2.fr/">bpp1.9-numcalc</a></td><td><a href="./bpp1.9-numcalc/">1.8.0-1</a></td><td>Bio++ 1.9 NumCalc Library</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://biopp.univ-montp2.fr/">bpp1.9-phyl</a></td><td><a href="./bpp1.9-phyl/">1.9.0-1</a></td><td>Bio++ 1.9 Phyl Library</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="http://biopp.univ-montp2.fr/">bpp1.9-seq</a></td><td><a href="./bpp1.9-seq/">1.7.0-1</a></td><td>Bio++ 1.9 Seq Library</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://biopp.univ-montp2.fr/">bpp1.9-utils</a></td><td><a href="./bpp1.9-utils/">1.5.0-1</a></td><td>Bio++ 1.9 Utils Library</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="https://github.com/BioPP/bpp-seq/">bppsuite</a></td><td><a href="./bppsuite/">2.4.1-1</a></td><td>Bio++ Suite</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://bio-bwa.sourceforge.net/">bwa</a></td><td><a href="./bwa/">0.7.17-1</a></td><td>Burrows-Wheeler Aligner</td><td>2022-07-21</td></tr>
   <tr class="even"><td><a href="https://faculty.sites.iastate.edu/xqhuang/cap3-assembly-program">cap3</a></td><td><a href="./cap3/">20091029-1</a></td><td>CAP3 Assembly Program</td><td>2022-07-21</td></tr>
   <tr class="odd"><td><a href="http://bioinf.gen.tcd.ie/~faresm/software/software.html">caps-coev</a></td><td><a href="./caps-coev/">2.0-2</a></td><td>Coevolution Analysis using Protein Sequences</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="http://ruby.chemie.uni-freiburg.de/~martin/chemtool/">chemtool</a></td><td><a href="./chemtool/">1.6.14-1</a></td><td>Chemical Molecule Drawing Program</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://www.clustal.org/omega/">clustalo</a></td><td><a href="./clustalo/">1.2.4-1</a></td><td>Clustal Omega</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="http://www.clustal.org/">clustalw</a></td><td><a href="./clustalw/">2.1-1</a></td><td>ClustalW</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://www.clustal.org/">clustalx</a></td><td><a href="./clustalx/">2.1-2</a></td><td>ClustalW frontend for X11</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="https://web.rniapps.net/compnet/">compnet</a></td><td><a href="./compnet/">1.0-1</a></td><td>Comparison of multiple interaction networks</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://cufflinks.cbcb.umd.edu/">cufflinks</a></td><td><a href="./cufflinks/">2.2.1-1</a></td><td>A reference-guided assembler for RNA-Seq experiments</td><td>2022-07-22</td></tr>
   <tr class="even"><td><a href="https://pypi.python.org/pypi/cutadapt/">cutadapt1</a></td><td><a href="./cutadapt1/">1.14-1</a></td><td>Trim adapters from high-throughput sequencing reads</td><td>2022-07-22</td></tr>
   <tr class="odd"><td><a href="https://sourceforge.net/projects/data2fcs/">data2fcs</a></td><td><a href="./data2fcs/">1.0-1</a></td><td>Data2FCS and FCS2Data</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="https://www.gnu.org/software/datamash/">datamash</a></td><td><a href="./datamash/">1.7-1</a></td><td>Basic numeric, textual and statistical operations</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="https://dendropy.org/">dendropy</a></td><td><a href="./dendropy/">4.5.1-1</a></td><td>A Phylogenetic Computing Library</td><td>2022-07-18</td></tr>
   <tr class="even"><td><a href="http://emboss.sourceforge.net/embassy/">embassy</a></td><td><a href="./embassy/">6.6.0-3</a></td><td>EMBOSS associated software</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://emboss.sourceforge.net/">emboss</a></td><td><a href="./emboss/">6.6.0-2</a></td><td>The European Molecular Biology Open Software Suite</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="https://www.ebi.ac.uk/about/vertebrate-genomics/software/exonerate">exonerate</a></td><td><a href="./exonerate/">2.4.0-1</a></td><td>Exonerate: a generic tool for sequence alignment</td><td>2022-04-19</td></tr>
   <tr class="odd"><td>facsdir</td><td><a href="./facsdir/">0.1-1</a></td><td>Desktop directory shared by FACS applications</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/">fastqc</a></td><td><a href="./fastqc/">0.11.9-1</a></td><td>A quality control tool for high throughput sequence data</td><td>2022-07-22</td></tr>
   <tr class="odd"><td><a href="https://sourceforge.net/projects/fcsalyzer">fcsalyzer</a></td><td><a href="./fcsalyzer/">0.9.22-1</a></td><td>Flow cytometry analysis program</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="https://fiji.sc/">fiji</a></td><td><a href="./fiji/">20220414_1745-1</a></td><td>Fiji is Just ImageJ, with extras</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="https://digitalworldbiology.com/FinchTV">finchtv</a></td><td><a href="./finchtv/">1.3.1-1</a></td><td>Geospiza's FinchTV is the popular way to view DNA sequence traces</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="https://github.com/pn2200/g3data">g3data</a></td><td><a href="./g3data/">1.5.4-1</a></td><td>A tool for extracting data from graphs</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://molevol.cmima.csic.es/castresana/Gblocks.html">gblocks</a></td><td><a href="./gblocks/">0.91b-1</a></td><td>Select blocks of evolutionarily conserved sites</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="http://download.savannah.gnu.org/releases/gchemutils">gchemutils</a></td><td><a href="./gchemutils/">0.12.13-2</a></td><td>GNOME Chemistry Utilities</td><td>2022-04-30</td></tr>
   <tr class="odd"><td><a href="https://github.com/ginggs/gelemental">gelemental</a></td><td><a href="./gelemental/">2.0.1-1</a></td><td>Periodic table</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="https://www.math.wustl.edu/~sawyer/geneconv/">geneconv</a></td><td><a href="./geneconv/">1.81a-1</a></td><td>Statistical Tests for Detecting Gene Conversion</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://genometools.org/">genometools</a></td><td><a href="./genometools/">1.6.2-1</a></td><td>A collection of bio-informatics tools</td><td>2022-07-06</td></tr>
   <tr class="even"><td><a href="http://www.bioinformatics.org/genpak/">genpak</a></td><td><a href="./genpak/">0.26-1</a></td><td>Utilities to manipulate DNA sequences</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="https://gephi.org/">gephi</a></td><td><a href="./gephi/">0.9.6-1</a></td><td>The Open Graph Viz Platform</td><td>2022-07-06</td></tr>
   <tr class="even"><td><a href="http://gp2fasta.netmark.pl/">gp2fasta</a></td><td><a href="./gp2fasta/">0.2-1</a></td><td>Convert gp files to fasta</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://guidance.tau.ac.il/">guidance</a></td><td><a href="./guidance/">2.02-1</a></td><td>GUIDe tree based AligNment ConfidencE</td><td>2022-04-24</td></tr>
   <tr class="even"><td><a href="http://eddylab.org/software">hmmer</a></td><td><a href="./hmmer/">3.3.2-1</a></td><td>Hidden Markov Model profile</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="https://github.com/veg/hyphy/">hyphy</a></td><td><a href="./hyphy/">2.5.39-1</a></td><td>Hypothesis Testing using Phylogenies</td><td>2022-07-13</td></tr>
   <tr class="even"><td><a href="http://ipc.netmark.pl/">iepcalc</a></td><td><a href="./iepcalc/">1.0-1</a></td><td>Isoelectric pint calculator</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://software.broadinstitute.org/software/igv/home">igv</a></td><td><a href="./igv/">2.13.1-1</a></td><td>Integrative Genomics Viewer</td><td>2022-07-06</td></tr>
   <tr class="even"><td><a href="https://github.com/jkbonfield/io_lib">io_lib</a></td><td><a href="./io_lib/">1.14.14-1</a></td><td>Provides a trace file reading interface</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://www.iqtree.org/">iqtree</a></td><td><a href="./iqtree/">1.6.12-1</a></td><td>Efficient software for phylogenomic inference</td><td>2022-07-22</td></tr>
   <tr class="even"><td><a href="http://www.jalview.org/">jalview</a></td><td><a href="./jalview/">2.11.2.0-1</a></td><td>A Sequence Alignment Editor and Viewer</td><td>2022-07-06</td></tr>
   <tr class="odd"><td><a href="http://pbil.univ-lyon1.fr/software/lalnview.html">lalnview</a></td><td><a href="./lalnview/">3.0-1</a></td><td>Visualize local alignments</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="https://mafft.cbrc.jp/alignment/software/">mafft</a></td><td><a href="./mafft/">7.490-1</a></td><td>A multiple sequence alignment program</td><td>2022-07-06</td></tr>
   <tr class="odd"><td><a href="https://micans.org/mcl/">mcl</a></td><td><a href="./mcl/">14_137-1</a></td><td>Markov Cluster Algorithm</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="http://www.megasoftware.net/">mega</a></td><td><a href="./mega/">11.0.11-1</a></td><td>Molecular Evolutionary Genetics Analysis</td><td>2022-07-06</td></tr>
   <tr class="odd"><td><a href="https://meme-suite.org/meme/db/gomo">meme-db-gomo</a></td><td><a href="./meme-db-gomo/">3.2-1</a></td><td>GOMo Databases to be used with MEME Suite</td><td>2022-07-13</td></tr>
   <tr class="even"><td><a href="https://meme-suite.org/db/motifs">meme-db-motif</a></td><td><a href="./meme-db-motif/">12.23-1</a></td><td>Motif Databases to be used with MEME Suite</td><td>2022-07-13</td></tr>
   <tr class="odd"><td><a href="https://meme-suite.org/db/tgene">meme-db-tgene</a></td><td><a href="./meme-db-tgene/">1.0-1</a></td><td>T-Gene Databases to be used with MEME Suite</td><td>2022-07-13</td></tr>
   <tr class="even"><td><a href="https://meme-suite.org/">meme-suite</a></td><td><a href="./meme-suite/">5.4.1-1</a></td><td>Motif-based sequence analysis tools</td><td>2022-07-13</td></tr>
   <tr class="odd"><td><a href="https://github.com/NBISweden/MrBayes">mrbayes</a></td><td><a href="./mrbayes/">3.2.7a-1</a></td><td>Bayesian inference of phylogeny</td><td>2022-07-21</td></tr>
   <tr class="even"><td><a href="https://msa.sbc.su.se/cgi-bin/msa.cgi">mumsa</a></td><td><a href="./mumsa/">1.0-1</a></td><td>Assessment of the quality of multiple sequence alignments</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://www.drive5.com/muscle/">muscle</a></td><td><a href="./muscle/">3.8.1551-1</a></td><td>MUltiple Sequence Comparison by Log-Expectation</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="ftp://ftp.ncbi.nih.gov/blast/executables/legacy.NOTSUPPORTED/">ncbi-blast</a></td><td><a href="./ncbi-blast/">2.2.26-1</a></td><td>BLAST: Basic Local Alignment Search Tool</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="https://blast.ncbi.nlm.nih.gov/">ncbi-blast-plus</a></td><td><a href="./ncbi-blast-plus/">2.13.0-1</a></td><td>BLAST+ Command Line Applications</td><td>2022-07-06</td></tr>
   <tr class="even"><td><a href="ftp://ftp.ncbi.nlm.nih.gov/blast/executables/rmblast/">ncbi-rmblastn</a></td><td><a href="./ncbi-rmblastn/">2.2.28-1</a></td><td>NCBI RMBlast</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://www.bork.embl.de/pal2nal/">pal2nal</a></td><td><a href="./pal2nal/">14.1-1</a></td><td>Create a codon-based DNA alignment</td><td>2022-07-13</td></tr>
   <tr class="even"><td><a href="http://abacus.gene.ucl.ac.uk/software/paml.html">paml</a></td><td><a href="./paml/">4.9j-1</a></td><td>PAML: Phylogenetic Analysis by Maximum Likelihood</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://abacus.gene.ucl.ac.uk/software/paml.html">pamlx</a></td><td><a href="./pamlx/">1.3.1-1</a></td><td>pamlX: the graphical user interface for PAML</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="https://github.com/owenjm/perlprimer/">perlprimer</a></td><td><a href="./perlprimer/">f415ea6-1</a></td><td>Design primers for PCR and sequencing</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="https://github.com/stephaneguindon/phyml/">phyml</a></td><td><a href="./phyml/">3.3.20220408-1</a></td><td>Phylogenetic estimation using Maximum Likelihood</td><td>2022-04-23</td></tr>
   <tr class="even"><td><a href="https://github.com/jydu/physamp">physamp</a></td><td><a href="./physamp/">1.1.0-1</a></td><td>Phylogenetic Sampling and Optimization of Sequence Alignments</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://wasabiapp.org/software/prank/">prank</a></td><td><a href="./prank/">170427-1</a></td><td>A probabilistic multiple alignment program</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="https://cme.h-its.org/exelixis/web/software/raxml/index.html">raxml</a></td><td><a href="./raxml/">8.2.12-1</a></td><td>RAxML - Randomized Axelerated Maximum Likelihood</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://www.repeatmasker.org/">repeatmasker</a></td><td><a href="./repeatmasker/">4.1.2_p1-1</a></td><td>Screen DNA sequence for interspersed repeats</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="https://www.rstudio.com/">rstudio-desktop</a></td><td><a href="./rstudio-desktop/">2022.02.3+492-1</a></td><td>RStudio is an integrated development environment (IDE) for R</td><td>2022-07-06</td></tr>
   <tr class="odd"><td>samtools0</td><td><a href="./samtools0/">0.1.19-1</a></td><td></td><td>2022-07-21</td></tr>
   <tr class="even"><td><a href="http://doua.prabi.fr/software/seaview">seaview</a></td><td><a href="./seaview/">5.0.5-1</a></td><td>Multiple sequence alignment and phylogeny GUI</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://informagen.com/SA/">seq-ana</a></td><td><a href="./seq-ana/">1.7.2-1</a></td><td>Biologic sequences analysis</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="https://bioinf.shenwei.me/seqkit/">seqkit</a></td><td><a href="./seqkit/">2.2.0-1</a></td><td>Atoolkit for FASTA/Q file manipulation</td><td>2022-07-06</td></tr>
   <tr class="odd"><td><a href="https://pcingola.github.io/SnpEff/">snpeff</a></td><td><a href="./snpeff/">4_3t-1</a></td><td>Genetic variant annotation and effect prediction toolbox</td><td>2022-07-22</td></tr>
   <tr class="even"><td><a href="http://cab.spbu.ru/software/spades/">spades</a></td><td><a href="./spades/">3.15.4-1</a></td><td>A single-cell genome assembler</td><td>2022-07-22</td></tr>
   <tr class="odd"><td><a href="https://www.ncbi.nlm.nih.gov/spidey/index.html">spidey</a></td><td><a href="./spidey/">20060601-1</a></td><td>mRNA-to-genomic alignment</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="ftp://ftp.pasteur.fr/pub/gensoft/projects/squizz/">squizz</a></td><td><a href="./squizz/">0.99d-1</a></td><td>A sequence/alignment format checker and converter</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://staden.sourceforge.net/">staden</a></td><td><a href="./staden/">2.0.0b11-1</a></td><td>DNA sequence assembly, editing, and analysis tools</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="http://ccb.jhu.edu/software/stringtie/">stringtie</a></td><td><a href="./stringtie/">2.2.1-1</a></td><td>Transcript assembly and quantification for RNA-Seq</td><td>2022-07-22</td></tr>
   <tr class="odd"><td><a href="http://www.tcoffee.org/">t_coffee</a></td><td><a href="./t_coffee/">0125a58-1</a></td><td>A multiple sequence alignment program</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="http://samtools.sourceforge.net/tabix.shtml">tabix0</a></td><td><a href="./tabix0/">0.2.6-1</a></td><td>Legacy version of Tabix</td><td>2022-07-22</td></tr>
   <tr class="odd"><td><a href="http://ppuigbo.me/programs/topd/">topd</a></td><td><a href="./topd/">4.6-1</a></td><td>TOPD/FMTS is a new software for comparing phylogenetic trees</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="http://crystal.scb.uwa.edu.au/charlie/software/topdraw/">topdraw</a></td><td><a href="./topdraw/">sep2002-1</a></td><td>A sketchpad for protein topology diagrams</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="https://github.com/Ensembl/treebest">treebest</a></td><td><a href="./treebest/">347fa82-1</a></td><td>Tree Building guidEd by Species Tree</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="https://project.inria.fr/treerecs/">treerecs</a></td><td><a href="./treerecs/">1.2-1</a></td><td>Treerecs: a tree reconciliation tool</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="https://github.com/rdmpage">treeviewx</a></td><td><a href="./treeviewx/">0.5.1-1</a></td><td>Program to display phylogenetic trees</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="https://tandem.bu.edu/trf/trf.html">trf</a></td><td><a href="./trf/">4.09.1-1</a></td><td>Tandem repeats finder: a program to analyze DNA sequences</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://ugene.net/">ugene</a></td><td><a href="./ugene/">43.0-1</a></td><td>A unified bioinformatics toolkit</td><td>2022-07-12</td></tr>
   <tr class="even"><td><a href="http://www.frankbattye.com.au/freesoftware.html">wehi-cellsymphony</a></td><td><a href="./wehi-cellsymphony/">20130721-1</a></td><td>Audio representation of flow cytometric data</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="http://www.frankbattye.com.au/freesoftware.html">wehi-deid</a></td><td><a href="./wehi-deid/">20130720-1</a></td><td>Flow cytometry De-Identifier</td><td>2022-04-19</td></tr>
   <tr class="even"><td><a href="http://www.frankbattye.com.au/freesoftware.html">wehi-hackit</a></td><td><a href="./wehi-hackit/">20050203-1</a></td><td>Clip a stream of cells from flow cytometry files</td><td>2022-04-19</td></tr>
   <tr class="odd"><td><a href="https://pypi.python.org/pypi/xopen">xopen0</a></td><td><a href="./xopen0/">0.2.1-1</a></td><td>A xopen function that can deal with compressed files</td><td>2022-07-22</td></tr>
  </table>
  <p><b>107 ports</b></p>
  <p><i>Generated by portspage 1.0.5 on 2022-07-23 06:45:32 UTC.</i></p>
 </body>
</html>
