From 958d12bdd35e7f6b0ca4d82fd95337d750535798 Mon Sep 17 00:00:00 2001
From: Paul Wise <pabs3@bonedaddy.net>
Date: Fri, 23 Feb 2024 06:05:45 +0800
Subject: [PATCH] Add channel context menu item to close the channel

---
 src/fe-gtk/menu.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/src/fe-gtk/menu.c b/src/fe-gtk/menu.c
index 76bc3906c..7b31722c4 100644
--- a/src/fe-gtk/menu.c
+++ b/src/fe-gtk/menu.c
@@ -989,6 +989,18 @@ menu_chan_cycle (GtkWidget * menu, char *chan)
 	}
 }
 
+static void
+menu_chan_close (GtkWidget * menu, char *chan)
+{
+	char tbuf[256];
+
+	if (current_sess)
+	{
+		g_snprintf (tbuf, sizeof tbuf, "doat %s close", chan);
+		handle_command (current_sess, tbuf, FALSE);
+	}
+}
+
 static void
 menu_chan_part (GtkWidget * menu, char *chan)
 {
@@ -1053,6 +1065,8 @@ menu_chanmenu (struct session *sess, GdkEventButton * event, char *chan)
 		if (chan_session != current_sess)
 			menu_quick_item_with_callback (menu_chan_focus, _("Focus Channel"), menu,
 													 str_copy);
+		menu_quick_item_with_callback (menu_chan_close, _("Close Channel"), menu,
+												 str_copy);
 		menu_quick_item_with_callback (menu_chan_part, _("Part Channel"), menu,
 												 str_copy);
 		menu_quick_item_with_callback (menu_chan_cycle, _("Cycle Channel"), menu,
